src/main/java/bio/terra/app/usermetrics/UserLoggingMetrics.java

```diff
@@ -0,0 +1,25 @@
+package bio.terra.app.usermetrics;
+
+import java.util.HashMap;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserLoggingMetrics {
+
+  private static final ThreadLocal<HashMap<String, Object>> metrics =
+      ThreadLocal.withInitial(() -> new HashMap<>());
+
+  public HashMap<String, Object> get() {
+    return metrics.get();
+  }
+
+  public void set(String key, Object value) {
```

Comment:
I think we'll want some javadocs here so we know what this class/methods does/do